runtime/common/src/routing.rs

```diff
@@ -78,3 +80,40 @@ impl<Runtime: pallet_evm::Config> PreConditions<(H160, H256)> for EvmAccountCode
 		BlakeTwo256::hash_of(&code) == contract_hash
 	}
 }
+
+pub struct MessageSerializer<Sender, Receiver>(PhantomData<(Sender, Receiver)>);
```

Comment:
Epic!

Reply:
I agree with Cosmin! The result was super cool!